build.assets/Dockerfile-centos7

```diff
@@ -48,25 +57,27 @@ RUN git clone --depth=1 https://github.com/git/git.git -b v2.42.0 && \
     make -j"$(nproc)" all && \
     DESTDIR=/opt/git make install"
 
-# Create an alias to the assets image. Ref: https://github.com/docker/for-mac/issues/2155
-ARG BUILDARCH
-FROM ghcr.io/gravitational/teleport-buildbox-centos7-assets:teleport15-${BUILDARCH} AS teleport-buildbox-centos7-assets
-
 ## LIBFIDO2 ###################################################################
 
 # Build libfido2 separately for isolation, speed and flexibility.
-FROM centos:7 AS libfido2
+FROM --platform=$BUILDPLATFORM centos:7 AS libfido2
```

Comment:
I notice we have `--platform=$BUILDPLATFORM` for all of these besides the final one:

https://github.com/gravitational/teleport/blob/23b28fa065a265ecb9ee3b3374d195bc02f833eb/build.assets/Dockerfile-centos7#L247

Is this intentional? I also don't see it being used anywhere, is it necessary?

Reply:
Yup, that's intentional. We want to use `$TARGETPLATFORM` here since it's the runtime image, which is the default. See https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/ for a good overview on how this works.